src/System.IO.Pipelines/tests/PipePoolTests.cs

```diff
@@ -266,6 +267,114 @@ public async Task OnReaderCompletedCalledAfterBlocksReturned()
             pipe.Reader.Complete();
         }
 
+        [Fact]
+        public async Task WritesToArrayPoolByDefault()
+        {
+            var pipe = new Pipe();
+            pipe.Writer.WriteEmpty(10);
+            await pipe.Writer.FlushAsync();
+            pipe.Writer.Complete();
+
+            ReadResult result = await pipe.Reader.ReadAsync();
+            Assert.Equal(10, result.Buffer.Length);
+
+            SequenceMarshal.TryGetReadOnlySequenceSegment(
+               result.Buffer,
+               out ReadOnlySequenceSegment<byte> start,
+               out int startIndex,
+               out ReadOnlySequenceSegment<byte> end,
+               out int endIndex);
+
+            var startSegment = (BufferSegment)start;
```

Comment:
```text
 PipePoolTests.cs(288,33): error CS0246: The type or namespace name 'BufferSegment' could not be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\System.IO.Pipelines\tests\System.IO.Pipelines.Tests.csproj]
```

It looks like we don't have `BufferSegment` on netstandard. Move tests to a partial class and only include it on nonnetstandard?
https://github.com/dotnet/corefx/blob/0c5594864387aef6d9f6c8842f8a9e476ef39cc3/src/System.IO.Pipelines/tests/System.IO.Pipelines.Tests.csproj#L32

Reply:
Yea I don't understand the error, trying to figure it out.